### PR TITLE
[Keyboard] add Symmetric70 keyboard prototype

### PR DESCRIPTION
**Pull request by mtei**
## Description

mtei's handmade prototype keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/symmetric70_proto/config.h [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0x0000
```
Comment (drashna) [inline keyboards/handwired/symmetric70_proto/config.h:24]:
```suggestion
#define PRODUCT_ID   0x2BE5
```

File: keyboards/handwired/symmetric70_proto/matrix.c.patch [hunk 2]
```diff
@@ -0,0 +1,37 @@
+--- quantum/matrix.c	2020-10-18 16:50:22.000000000 +0900
```
Comment (drashna) [inline keyboards/handwired/symmetric70_proto/matrix.c.patch:1]: Is this file needed?

File: keyboards/handwired/symmetric70_proto/symmetric70_proto.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "symmetric70_proto.h"
+
+void keyboard_post_init_user(void) {
```
Comment (drashna) [inline keyboards/handwired/symmetric70_proto/symmetric70_proto.c:19]:
```suggestion
__attribute__((weak)) void keyboard_post_init_user(void) {
```

or 
```suggestion
void keyboard_post_init_kb(void) {
```

File: keyboards/handwired/symmetric70_proto/readme.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# symmetric70_proto
+
+![symmetric70_proto](https://i.imgur.com/Br4pH9o.jpg)
```
Comment (fauxpark) [inline keyboards/handwired/symmetric70_proto/readme.md:3]:
```suggestion
![symmetric70_proto](https://i.imgur.com/Br4pH9ol.jpg)
```

File: keyboards/handwired/symmetric70_proto/readme.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+<img width="90%" alt="74HC157" src="https://user-images.githubusercontent.com/2170248/97360929-33765280-18e2-11eb-9eeb-791133e4d328.PNG">
```
Comment (fauxpark) [inline keyboards/handwired/symmetric70_proto/readme.md:4]: If you host this on imgur and add `l` to the filename (as above), you won't need the img tag here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/symmetric70_proto/matrix.c.patch (hunk 2 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/matrix.c.patch]: removed.

File: keyboards/handwired/symmetric70_proto/readme.md (hunk 5 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/readme.md]: Thanks.

File: keyboards/handwired/symmetric70_proto/config.h (hunk 1 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/config.h]: Thnaks.

File: keyboards/handwired/symmetric70_proto/matrix.c [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+    // initialize matrix state: all keys off
+    for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
+        raw_matrix[i] = 0;
+        matrix[i]     = 0;
+    }
+
+    debounce_init(MATRIX_ROWS);
+
+    matrix_init_quantum();
```
Comment (drashna) [inline keyboards/handwired/symmetric70_proto/matrix.c:210]: For matrix lite, this should be 
```suggestion
void matrix_init_custom(void) {
    // initialize key pins
    init_pins();

    // initialize matrix state: all keys off
    for (uint8_t i = 0; i < MATRIX_ROWS; i++) {
        raw_matrix[i] = 0;
        matrix[i]     = 0;
    }

```

File: keyboards/handwired/symmetric70_proto/matrix.c [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+    for (uint8_t current_col = 0; current_col < MATRIX_COLS; current_col++) {
+        changed |= read_rows_on_col(raw_matrix, current_col);
+    }
+#endif
+
+    debounce(raw_matrix, matrix, MATRIX_ROWS, changed);
+
+    matrix_scan_quantum();
+    return (uint8_t)changed;
+}
```
Comment (drashna) [inline keyboards/handwired/symmetric70_proto/matrix.c:232]:
```suggestion
bool matrix_scan_custom(matrix_row_t current_matrix[]) {
    bool changed = false;

#if defined(DIRECT_PINS) || (DIODE_DIRECTION == COL2ROW)
    // Set row, read cols
    for (uint8_t current_row = 0; current_row < MATRIX_ROWS; current_row++) {
        changed |= read_cols_on_row(current_matrix, current_row);
    }
#elif (DIODE_DIRECTION == ROW2COL)
    // Set col, read rows
    for (uint8_t current_col = 0; current_col < MATRIX_COLS; current_col++) {
        changed |= read_rows_on_col(current_matrix, current_col);
    }
#endif

    return changed;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/symmetric70_proto/matrix.c (hunk 6 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/matrix.c]: Thanks.

File: keyboards/handwired/symmetric70_proto/matrix.c (hunk 7 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/matrix.c]: Thanks

File: keyboards/handwired/symmetric70_proto/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    _BASE,
+    _FN
+};
+
+#define KC_FUNC MO(_FN)
+
+// clang-format off
+
+#define KC___ KC_TRNS
+#define KC_xx KC_NO
```
Comment (noroadsleft) [inline keyboards/handwired/symmetric70_proto/keymaps/default/keymap.c:29]: 2. I'm personally not a fan of the use of these sorts of keycode aliases, and would much prefer the QMK-native `_______` and `XXXXXXX` be used.

**Review by noroadsleft — COMMENTED** [review]
Just a couple of general comments.

1. The `default` and `default_mac` keymaps are the same, outside of the `default_mac` keymap including keycodes for `KC_LANG1` and `KC_LANG2`. I don't see why the `default_mac` keymap couldn't just *be* the default.

**Comment by mtei** [comment]
> 1. The `default` and `default_mac` keymaps are the same, outside of the `default_mac` keymap including keycodes for `KC_LANG1` and `KC_LANG2`. I don't see why the `default_mac` keymap couldn't just _be_ the default.

As far as I know, Apple's Japanese keyboard is the only keyboard that has the LANG1 and LANG2 keys. So I don't think keymaps with them are probably commonplace. That's why I didn't set default_mac to default.

File: keyboards/handwired/symmetric70_proto/keymaps/default/keymap.c (hunk 8 above)
Comment (mtei) [reply keyboards/handwired/symmetric70_proto/keymaps/default/keymap.c]: I don't like the keymap descriptions to be long in the horizontal direction.
However, I compromised a bit and changed it to use `KC_NO` and` ________`.

**Review by noroadsleft — APPROVED** [review]

**Review by drashna — APPROVED** [review]
